src/CCPNSimulatorPlugin.js

```diff
@@ -71,37 +89,42 @@ class CCPNSimulatorPlugin {
     }
 
     terminate () {
-        this.stop();
+        this.isLoaded = false;
+        this.isInitialized = false;
         this.ccpn.respawn();
     }
 
     getMarking () {
-        this.ccpn.sendElement('getMarking');
+        if (this.isLoaded) {
+            this.ccpn.sendElement('getMarking');
+        }
     }
 
     updateMarking (socket, netMarking) {
         if (!netMarking.placeState || !Array.isArray(netMarking.placeState)) {
             return;
         }
 
-        const newMarking = netMarking.placeState.map((placeState, index) => {
-            return {
+        const elements = netMarking.placeState.map((placeState, index) => {
+            const element = {
                 parentId: this.places[index].id,
-                text: placeState.token || '',
+                text: '',
                 type: 'pt:marking',
+            };
+            if (placeState.token && placeState.token.text) {
+                element.text = placeState.token.text;
             }
+            return element;
         });
+        const isHalted = netMarking.attributes
+            && netMarking.attributes.stopped === 'True';
```

Comment:
'True'...

Reply:
Could not get fast-xml-parser to parse this as boolean :disappointed: